### PR TITLE
Graph Module Extractors Interface

### PR DESCRIPTION
**Pull request by doctrino**
I ended up moving `rdf_file_to_graph` to the stores module. I think it does not belong with the extractors as it is not extracting anything but simply loading an already serialized graph.

### CODE REVIEW

File: cognite/neat/graph/extractors/_base.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from collections.abc import Iterable
+
+
+class BaseExtractor:
+    """This is the base class for all extractors. It defines the interface that
+    extractors must implement.
+    """
+
+    @abstractmethod
+    def extract(self) -> Iterable[tuple]:
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_base.py:11]: Extract should always return `Iterable[tuple[URIRef, URIRef, URIRef|Literal]]`. Checkout issue:
https://github.com/cognitedata/neat/issues/213

File: cognite/neat/graph/extractors/_graph_capturing_sheet.py [hunk 2]
```diff
@@ -59,7 +78,11 @@ def extract_triples_from_sheet(self, separator: str = ",", namespace: str | None
         """
         if self.filepath is None:
             raise ValueError("File path to the graph capturing sheet is not provided!")
-        return extract_graph_from_sheet(self.filepath, self.rules, separator, namespace)
+        graph_capturing_sheet = read_graph_excel_file_to_table_by_name(self.filepath)
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_graph_capturing_sheet.py:81]: I am wandering whether we should have an abstract `read` or `read_source` (or some similar name) method that would read source `object` from which we extract triples?

**Review by nikokaoja — APPROVED** [review]
See my two comments, otherwise LGTM

File: cognite/neat/graph/extractors/_graph_capturing_sheet.py (hunk 2 above)
Comment (doctrino) [reply cognite/neat/graph/extractors/_graph_capturing_sheet.py:82]: Lets keep the interface minimal for now until we see a strong need for it. You can always add methods to specific implementations.

**Pull request merged**